src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs

```diff
@@ -440,11 +440,29 @@ private string UFormatDateString(DateTime dateTime)
                             break;
 
                         case 'G':
-                            sb.Append("{0:yyyy}");
-                            break;
-
                         case 'g':
-                            sb.Append("{0:yy}");
+                            var isoWeek = ISOWeek.GetWeekOfYear(dateTime);
+                            var isoWeekYear = dateTime.Year;
+
+                            if (isoWeek == 53 && dateTime.Month == 1)
+                            {
+                                isoWeekYear--;
+                            }
+                            else if (isoWeek == 1 && dateTime.Month == 12)
+                            {
+                                isoWeekYear++;
+                            }
+
+                            if (UFormat[i] == 'g')
+                            {
+                                isoWeekYear %= 100;
+                                sb.Append(StringUtil.Format("{0:00}", isoWeekYear));
+                            }
+                            else
+                            {
+                                sb.Append(isoWeekYear);
```

Comment:
Based on my reading of the `G`/`Y` documentation, it's preferable for us to use a string formatter in every case here.

The logic in this `else` branch now depends on serialising the integer to its natural string representation, whereas `%G` in other libraries will left pad with zeroes. So the year `325` will be rendered as `0325` and the year `14` will be rendered as `0014`.

More than the actual logic difference, using a string formatter allows us to explicitly declare the intended formatting, rather than rely on .NET's own integer rendering logic.